modules/core/src/minmax.cpp

```diff
@@ -71,33 +71,760 @@ minMaxIdx_( const T* src, const uchar* mask, WT* _minVal, WT* _maxVal,
+                v_uint8x16 idx = idxStart, idxMin = none, idxMax = none;
+
+                int k = j;
+                size_t delta = startidx + j;
+
+                if ( !mask )
+                {
+                    for( ; k < std::min(len0, j + 15 * v_int8x16::nlanes); k += v_int8x16::nlanes )
+                    {
+                        v_int8x16 data = v_load(src + k);
+                        v_uint8x16 cmpMin = v_reinterpret_as_u8(data < valMin);
+                        v_uint8x16 cmpMax = v_reinterpret_as_u8(data > valMax);
+                        idxMin = v_select(cmpMin, idx, idxMin);
+                        idxMax = v_select(cmpMax, idx, idxMax);
+                        valMin = v_min(data, valMin);
+                        valMax = v_max(data, valMax);
+                        idx += inc;
+                    }
+                }
+                else
+                {
+                    for( ; k < std::min(len0, j + 15 * v_int8x16::nlanes); k += v_int8x16::nlanes )
+                    {
+                        v_int8x16 data = v_load(src + k);
+                        v_uint8x16 maskVal = v_load(mask + k) != v_setzero_u8();
+                        v_uint8x16 cmpMin = v_reinterpret_as_u8(data < valMin) & maskVal;
+                        v_uint8x16 cmpMax = v_reinterpret_as_u8(data > valMax) & maskVal;
+                        idxMin = v_select(cmpMin, idx, idxMin);
+                        idxMax = v_select(cmpMax, idx, idxMax);
+                        valMin = v_select(v_reinterpret_as_s8(cmpMin), data, valMin);
+                        valMax = v_select(v_reinterpret_as_s8(cmpMax), data, valMax);
+                        idx += inc;
+                    }
+                }
+
+                j = k;
+
+                minMaxIdx_reduce_s8( valMin, valMax, idxMin, idxMax, none, minVal, maxVal,
+                                     minIdx, maxIdx, delta );
+            }
+            while ( j < len0 );
+        }
+
+        minMaxIdx_finish( src, mask, minval, maxval, minidx, maxidx, minVal, maxVal,
+                          minIdx, maxIdx, len, startidx, j );
+    }
+    else
+    {
+        minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx);
+    }
+#else
+    minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx );
+#endif
+}
 
 static void minMaxIdx_16u(const ushort* src, const uchar* mask, int* minval, int* maxval,
                           size_t* minidx, size_t* maxidx, int len, size_t startidx )
-{ minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx ); }
+{
+#if CV_SIMD128
+    if ( len >= v_uint16x8::nlanes )
+    {
+        int j, len0;
+        int minVal, maxVal;
+        size_t minIdx, maxIdx;
+
+        minMaxIdx_init( src, mask, minval, maxval, minidx, maxidx, minVal, maxVal, minIdx, maxIdx,
+                        (int)0, (int)USHRT_MAX, v_uint16x8::nlanes, len, startidx, j, len0 );
+
+        if ( len0 - j >= v_uint16x8::nlanes )
```

Comment:
Please use this form: `j <= len0 - _uint16x8::nlanes` (used in many vectorized loops)